HT_BinaryTree/main.cpp

```diff
@@ -0,0 +1,17 @@
+#include "functions.h"
+
+
+int main()
+{
+    node *head;
+    head = new node(1,nullptr,nullptr);
+    head->left = new node(2,nullptr,nullptr);
+    head->right = new node(3,nullptr,nullptr);
+    head->left->left = new node(4,nullptr,nullptr);
+    head->left->right = new node(5,nullptr,nullptr);
+    head->right->left = new node(6,nullptr,nullptr);
+    head->right->right = new node(7,nullptr,nullptr);
```

Comment:
you'll have the memory leak here

Reply:
as i get it leak due to nullptr? changed to zero

Reply:
where is the delete?